src/session.rs

```diff
@@ -456,10 +461,13 @@ impl Stream for Session {
             }
         }
 
-        let rxh = self.state.rx_heartbeat_timeout
+        let rxh = self
+            .state
+            .rx_heartbeat_timeout
             .as_mut()
             .map(|t| t.poll())
-            .unwrap_or(Ok(Async::NotReady))?;
+            .unwrap_or(Ok(Async::NotReady))
+            .map_err(|_e| IoError::new(ErrorKind::Other, "timer"))?;
```

Comment:
I'm not very familiar with `tokio_timer`. Does it not provide an idiomatic return value to represent timeouts?

Reply:
There are two kinds of error, `shutdown` when there's no executor and `at_capacity` when timeout delay is too large to be handled (longer than 2 years in default timer).
`tokio-timer` has it's own error type, but it should be wrapped to `std::io::Error` because `Session::Error` is `std::io::Error`. We can change error string more specifically to help troubleshooting.